MachRegister/base_registers/x86.cpp

```diff
@@ -0,0 +1,48 @@
+#include "basereg_check.h"
+#include "registers/x86_regs.h"
+
+int main() {
+  BASEREG_CHECK(x86::ax, x86::eax);
+  BASEREG_CHECK(x86::eax, x86::eax);
+
+  BASEREG_CHECK(x86::sp, x86::esp);
+  BASEREG_CHECK(x86::esp, x86::esp);
```

Comment:
check byte-sized register to base and also check highest gpr register for numbering problems:

```suggestion
  BASEREG_CHECK(x86::al, x86::eax);
  BASEREG_CHECK(x86::ah, x86::eax);
  BASEREG_CHECK(x86::ax, x86::eax);
  BASEREG_CHECK(x86::eax, x86::eax);

  BASEREG_CHECK(x86::sp, x86::esp);
  BASEREG_CHECK(x86::esp, x86::esp);
  
  BASEREG_CHECK(x86::di, x86::edi);
  BASEREG_CHECK(x86::edi, x86::edi);
```